File: app/src/main/java/eu/toma/dev/playground/warcraft/mounts/di/MountComponent.java
package eu.toma.dev.playground.warcraft.mounts.di;

import dagger.Component;
import eu.toma.dev.playground.rx_example_one.RxOneActivity;
import eu.toma.dev.playground.warcraft.di.WoWcomponent;
import eu.toma.dev.playground.warcraft.mounts.MountsActivity;
import eu.toma.dev.playground.warcraft.mounts.all.AllMountsFragment;
import eu.toma.dev.playground.warcraft.mounts.details.MountDetailsFragment;

@Mounts
@Component(dependencies = WoWcomponent.class)
public interface MountComponent
{
    void inject(AllMountsFragment allMounts);

    void inject(MountDetailsFragment mountDetailsFragment);

    void inject(MountsActivity mountsActivity);
}

File: app/src/main/kotlin/warcraft/mounts/di/MountComponent.kt
package warcraft.mounts.di

import dagger.Component
import warcraft.di.WoWcomponent
import warcraft.mounts.MountsActivity
import warcraft.mounts.all.AllMountsFragment
import warcraft.mounts.details.MountDetailsFragment

@Mounts
@Component(dependencies = arrayOf(WoWcomponent::class))
interface MountComponent {
    fun inject(allMounts: AllMountsFragment)

    fun inject(mountDetailsFragment: MountDetailsFragment)

    fun inject(mountsActivity: MountsActivity)
}
